### ISSUE TITLE
Redeclaration of char8_t when using avr-g++ 9.2 when running with C++20

### ISSUE THREAD

**Issue by jputcu** [ETLCPP/etl#726]
Using Zak Kemble 9.2 AVR-GCC compiler and activating C++20 I get the following errors in `etl/20.35.14`:
~~~~
include/etl/platform.h:331:20: error: redeclaration of C++ built-in type 'char8_t' [-fpermissive]
  331 |     typedef int8_t char8_t;
      |                    ^~~~~~~
~~~~
When running in C++17 I don't get that error.

**Comment by jputcu** [comment]
I see that is is probably fixed in `20.36`, which I currently couldn't use due to another compilation error.

**Comment by jwellbelove** [comment]
Can you open an issue on the compilation error that you get in `20.36`?
